src/nvim/lua/executor.c

```diff
@@ -1649,6 +1649,15 @@ bool nlua_is_deferred_safe(void)
 void ex_lua(exarg_T *const eap)
   FUNC_ATTR_NONNULL_ALL
 {
+  if (eap->addr_count > 0 || *eap->arg == NUL) {
+    if (eap->addr_count > 0 && *eap->arg == NUL) {
+      cmd_source_buffer(eap, true);
+    } else {
+      semsg(_(e_invarg2), "one of {chunk} and [count] is required");
```

Comment:
Let's use the same phrasing as https://github.com/neovim/neovim/blob/4e59422e1d4950a3042bad41a7b81c8db4f8b648/src/nvim/eval/typval.h#L441

```suggestion
      semsg(_(e_invarg2), "{chunk} or [count] is required");
```

Reply:
I don't think this is a good idea, because both `{chunk}` and `{range}` can be specified, which needs to be rejected.

Reply:
Go with the previous phrasing then? Or just

    either {chunk} or {range} is required

Reply:
Or do we need separate errors:

    {chunk} or {range} required

and

    only one of {chunk} and {range} allowed

Reply:
`exactly one of`?

Reply:
Works for me. Is there no precedent for this kind of message?

Reply:
Hmm, `nvim_create_autocmd` has two separate error messages for mutually exclusive `command` and `callback`.